poetry/utils/extras.py

```diff
@@ -14,7 +14,7 @@ def get_extra_package_names(
     packages: Sequence["Package"],
     extras: Mapping[str, List[str]],
     extra_names: Sequence[str],
-) -> Iterator[str]:
+) -> Iterable[str]:
```

Comment:
@abn 
this too is a *correctness* bug. the returned value of this function is *not* an iterator, but it *is* a type than can be converted *into* an iterator. That's the distinction between `Iterator` and `Iterable`